lib/polly_stubs.c

```diff
@@ -119,4 +119,39 @@ caml_polly_wait(value val_epfd, value val_max, value val_timeout, value val_f)
 	CAMLreturn(Val_int(ready));
 }
 
+CAMLprim value
+caml_polly_wait_fold(value val_epfd, value val_max, value val_timeout,
+		     value val_init, value val_f)
+{
+	CAMLparam5(val_epfd, val_max, val_timeout, val_init, val_f);
+	value args[4];
```

Comment:
I don't think that this is safe without CAMLlocal.

Reply:
You can use `CAMLlocalN`.

Reply:
Actually, no: https://github.com/ocaml/ocaml/issues/11045 :)

Reply:
I'm not sure/ I looked at the OCaml code:

* https://github.com/ocaml/ocaml/blob/trunk/testsuite/tests/callback/test1_.c#L41-L51

Reply:
Answered on the OCaml forum: https://discuss.ocaml.org/t/camllocal-for-caml-callbackn-arg-array-required/11012.

Reply:
I think that it is alright, because `caml_callbackN` registers `args` as roots, so that inside of it, all values are protected. Also, between the assignment of any of the `args` fields and the `caml_callbackN` calls, there are is no code that does OCaml allocations and may trigger the GC.